### PR TITLE
Feat installer rework 2

### LINKED ISSUE #1200: Debian installer is broken
**7.x**: broken https://beats-ci.elastic.co/blue/organizations/jenkins/e2e-tests%2Fe2e-testing-mbp%2F7.x/detail/7.x/545/pipeline
**master**: broken too https://beats-ci.elastic.co/blue/organizations/jenkins/e2e-tests%2Fe2e-testing-mbp%2Fmaster/detail/master/908/pipeline

>[2021-05-26T05:39:10.992Z] time="2021-05-26T05:39:10Z" level=fatal msg="Sorry, we currently do not support this installer" image=debian installer=tar

@adam-stokes I think we broke the debian builds with https://github.com/elastic/e2e-testing/pull/1163. When reviewing it, I noticed we changed the code to retrieve an installer in `elasticagent.go#GetElasticAgentInstaller`:

```golang
// TODO: convert to new installer code
	if "docker" == image && "default" == installerType {
		installer, err = newDockerInstaller(false, version)
	} else if "docker" == image && "ubi8" == installerType {
		installer, err = newDockerInstaller(true, version)
	} else {
		log.WithFields(log.Fields{
			"image":     image,
			"installer": installerType,
		}).Fatal("Sorry, we currently do not support this installer")
		return ElasticAgentInstaller{}
	}
```

I'm thinking about a revert for that commit, leaving the team the responsibility to rework on top of the revert.

### PR DESCRIPTION
**Pull request by adam-stokes**
Continuation of #1163 

Fixes #1200 

This was tested against both debian/centos using the same scripts from .ci/

### CODE REVIEW

File: e2e/_suites/fleet/stand-alone.go [hunk 1]
```diff
@@ -102,14 +104,14 @@ func (fts *FleetTestSuite) startStandAloneAgent(image string, flavour string, en
...
 		// load the docker images that were already:
 		// a. downloaded from the GCP bucket
 		// b. fetched from the local beats binaries
-		dockerInstaller := installer.GetElasticAgentInstaller("docker", image, common.BeatVersion, deployedAgentsCount)
-
-		dockerInstaller.PreInstallFn()
-
+		agentService := deploy.NewServiceRequest(common.ElasticAgentServiceName)
+		dockerInstaller, _ := installer.Attach(fts.deployer, agentService, "docker")
+		dockerInstaller.Preinstall()
```
Comment (adam-stokes) [inline e2e/_suites/fleet/stand-alone.go:114]: @mdelapenya After adding the docker installer interface this is the main culprit of #1200

**Comment by adam-stokes** [comment]
Same flaky test failure as before

**Comment by adam-stokes** [comment]
jenkins please test this

**Review by mdelapenya — APPROVED** [review]
@adam-stokes I've tried this PR with different combinations, including Debian and Ubi8 images:

### current snapshot
```shell
TAGS="backend_processes" TIMEOUT_FACTOR=3 LOG_LEVEL=TRACE BEATS_USE_CI_SNAPSHOT=false DEVELOPER_MODE=true make -C e2e/_suites/fleet functional-test

TAGS="stand_alone_agent" TIMEOUT_FACTOR=3 LOG_LEVEL=TRACE BEATS_USE_CI_SNAPSHOT=false DEVELOPER_MODE=true make -C e2e/_suites/fleet functional-test

TAGS="agent_endpoint_integration" TIMEOUT_FACTOR=3 LOG_LEVEL=TRACE BEATS_USE_CI_SNAPSHOT=false DEVELOPER_MODE=true make -C e2e/_suites/fleet functional-test
```

### Beats CI snapshot for `cec5906b013f4a99b29614509ac44d7e3e75dfe0`
The `cec5906b013f4a99b29614509ac44d7e3e75dfe0` commit represents the master branch in Beats:

```shell
TAGS="backend_processes" TIMEOUT_FACTOR=3 LOG_LEVEL=TRACE BEATS_USE_CI_SNAPSHOT=true GITHUB_CHECK_SHA1=cec5906b013f4a99b29614509ac44d7e3e75dfe0 DEVELOPER_MODE=true make -C e2e/_suites/fleet functional-test

TAGS="stand_alone_agent" TIMEOUT_FACTOR=3 LOG_LEVEL=TRACE BEATS_USE_CI_SNAPSHOT=true GITHUB_CHECK_SHA1=cec5906b013f4a99b29614509ac44d7e3e75dfe0 DEVELOPER_MODE=true make -C e2e/_suites/fleet functional-test

TAGS="agent_endpoint_integration" TIMEOUT_FACTOR=3 LOG_LEVEL=TRACE BEATS_USE_CI_SNAPSHOT=true GITHUB_CHECK_SHA1=cec5906b013f4a99b29614509ac44d7e3e75dfe0 DEVELOPER_MODE=true make -C e2e/_suites/fleet functional-test
```

**Pull request merged**